### PR TITLE
Micronaut Gradle Plugin to 4.1.0

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Review by melix — APPROVED** [review]

**Comment by sdelamo** [comment]
it seems updating the gradle plugin to 4.1.0 has broken the Windows Native CLI Generation: 

https://ge.micronaut.io/s/axlg2xskbp2jq

```
> Task :micronaut-cli:generateResourcesConfigFile FROM-CACHE
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char <> at index 3: D:amicronaut-startermicronaut-starterstarter-clibuildlibsmicronaut-cli-4.1.1-SNAPSHOT.jar

	at java.base@17.0.7/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
> Task :micronaut-cli:nativeCompile FAILED
	at java.base@17.0.7/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
[native-image-plugin] GraalVM Toolchain detection is disabled
	at java.base@17.0.7/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
[native-image-plugin] GraalVM location read from environment variable: GRAALVM_HOME
	at java.base@17.0.7/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
[native-image-plugin] Native Image executable path: C:\hostedtoolcache\windows\graalvm-ce-java17-windows\22.3.2\x64\graalvm-ce-java17-22.3.2\bin\native-image.cmd
	at java.base@17.0.7/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)

	at java.base@17.0.7/java.nio.file.Path.of(Path.java:147)
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
	at java.base@17.0.7/java.nio.file.Paths.get(Paths.java:69)

	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.ClasspathUtils.stringToClasspath(ClasspathUtils.java:52)
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.addCustomImageClasspath(NativeImage.java:1582)

	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.processClasspathArgs(DefaultOptionHandler.java:417)
For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.consume(DefaultOptionHandler.java:69)
22 actionable tasks: 11 executed, 11 from cache
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage$NativeImageArgsProcessor.apply(NativeImage.java:1512)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.consume(DefaultOptionHandler.java:204)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage$NativeImageArgsProcessor.apply(NativeImage.java:1512)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.processNativeImageArgs(NativeImage.java:1770)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.completeImageBuild(NativeImage.java:959)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.build(NativeImage.java:1417)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1387)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.main(NativeImage.java:1374)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':micronaut-cli:nativeCompile'.
> Process 'command 'C:\hostedtoolcache\windows\graalvm-ce-java17-windows\22.3.2\x64\graalvm-ce-java17-22.3.2\bin\native-image.cmd'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

BUILD FAILED in 3m 23s

Publishing build scan...
https://ge.micronaut.io/s/axlg2xskbp2jq
```

Any idea @melix ?

**Comment by melix** [comment]
@fniephaus I think you changed the Windows escaping strategy in 0.9.26, right?

**Comment by fniephaus** [comment]
Yes, and it will be fixed in the 0.9.27 release. The fix is included in https://github.com/graalvm/native-build-tools/pull/506.

**Comment by sdelamo** [comment]
> 

@melix I assume Micronaut Gradle Plugin 4.1.1 will contain 0.9.27 ?

**Comment by melix** [comment]
At this stage I don't know when 0.9.27 will be out.

**Comment by sdelamo** [comment]
> At this stage I don't know when 0.9.27 will be out.

@fniephaus do you know? 

@melix I think we are blocked to upgrade to 4.1.x Micronaut gradle plugin until this is fixed.

**Comment by melix** [comment]
Choose your plague...

**Comment by fniephaus** [comment]
You may be able to downgrade to `0.9.25` until `0.9.27` is out. We are working on a release, but it will take a few days.

**Comment by melix** [comment]
That's also why I'm saying "choose your plague", because 0.9.25 has other problems (older version of metadata, incompatible with Gradle 8.3 and Maven 3.9.x...

File: gradle/templates.versions.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ log4j = "2.19.0"
 log4j-slf4j = "2.17.2"
 micronaut-aot = "2.0.1"
 micronaut-build-plugins = "6.5.3"
-micronaut-gradle-plugin = "4.0.3"
+micronaut-gradle-plugin = "4.1.0"
```
Comment (sdelamo) [inline gradle/templates.versions.toml]:
```suggestion
micronaut-gradle-plugin = "4.1.1"
```

File: settings.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ pluginManagement {
     }
     plugins {
         id 'com.microsoft.azure.azurefunctions' version '1.12.1'
-        id 'io.micronaut.application' version '4.0.3'
+        id 'io.micronaut.application' version '4.1.0'
```
Comment (sdelamo) [inline settings.gradle]:
```suggestion
        id 'io.micronaut.application' version '4.1.1'
```

**Comment by fniephaus** [comment]
Native Build Tools `0.9.27` was released today, so this PR should be unblocked

**Comment by sdelamo** [comment]
@melix I updated the PR to 4.1.1. Failure seems to still happen. any idea?

**Comment by melix** [comment]
@sdelamo unfortunately it seems there's _another_ regression :(

https://github.com/graalvm/native-build-tools/issues/511

**Comment by sdelamo** [comment]
closed via https://github.com/micronaut-projects/micronaut-starter/pull/2144

**Pull request closed without merging**